src/components/UserDisplayName/UserDisplayName.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { oneOfType, string } from 'prop-types';
+import { propTypes } from '../../util/types';
+
+const UserDisplayName = props => {
+  const { user, intl, deletedUserDisplayName, bannedUserDisplayName } = props;
+  const hasAttributes = user && user.attributes;
+  const userIsDeleted = hasAttributes && user.attributes.deleted;
+  const userIsBanned = hasAttributes && user.attributes.banned;
+  const userHasProfile = hasAttributes && user.attributes.profile;
+
+  const deletedUserDisplayNameInUse = deletedUserDisplayName
+    ? deletedUserDisplayName
+    : intl.formatMessage({
+        id: 'UserDisplayName.deleted',
+      });
+
+  const bannedUserDisplayNameInUse = bannedUserDisplayName
+    ? bannedUserDisplayName
+    : intl.formatMessage({
+        id: 'UserDisplayName.banned',
+      });
+
+  return userIsDeleted ? (
```

Comment:
I think this would be more readable in a format:
```js
  const classes = classNames(rootClassName, className);
  return (
    <span className={classes}>
      {userHasProfile
        ? user.attributes.profile.displayName
        : userIsDeleted
        ? deletedUserDisplayNameInUse
        : userIsBanned
        ? bannedUserDisplayNameInUse
        : null}
    </span>
  );
```